src/lt/objs/sidebar/workspace.cljs

```diff
@@ -284,61 +291,132 @@
+           (object/raise this :rename.focus))
+  :blur (fn []
+          (object/raise this :rename.blur)))
+
+(object/object* ::rename-dialog
+                :tags #{:tree.rename}
+                :max-height 130
+                :origin nil
+                :renaming? true
+                :order 1
+                :behaviors #{::rename-focus ::rename-blur ::rename-submit ::rename-cancel ::rename-close! ::rename}
+                :init (fn [this origin]
+                        (let [t (if (files/dir? (:path @origin))
+                                  "folder"
+                                  "file")
+                              rel-path (:path @origin)]
+                          (object/merge! this {:origin origin})
+                           [:div.rename-dialog
+                            [:p "Enter path for " t ":"]
+                            (rename-input this rel-path)
+                            ])))
+
+(defn popup-same-name-error
+  [filename origin]
+  (let [t (if (object/has-tag? origin :workspace.folder)
+            "Folder"
+            "File")]
+    (popup/popup! {:header (str t " already exists.")
+                   :body (str t " " filename " already exists, you'll have to pick a different name.")
+                   :buttons [{:label "ok"
+                              :post-action (fn []
+                                             (object/raise origin :rename.cancel)
+                                             (object/raise origin :start-rename!))}]})))
+
+(defn popup-invalid-path-error
+  [invalid-path origin]
+  (popup/popup! {:header "Path does not exist."
+                 :body (str "Path '" invalid-path "' does not exist, you'll have to pick a different one.")
+                 :buttons [{:label "ok"
+                            :post-action (fn []
+                                           (object/raise origin :rename.cancel)
+                                           (object/raise origin :start-rename!))}]}))
+
+;; TODO: When moving file or folder into root folder (e.g., LightTable/untitled.txt)
+;; the file or folder does not appear until the folder is manually refreshed - all other deeper folders appear fine however
+
+(defn rename-file
+  [origin path neue]
+  ;; When moving a file, current-ws likes to be told about the change before
+  ;; the actual move so the editor tab will not close
+  (let [folder-root? (object/has-tag? origin :workspace.folder.root)
+        file-root?   (object/has-tag? origin :workspace.file.root)
+        old-parent-path  (files/parent path)
+        new-parent-path  (files/parent neue)]
+    (if (or folder-root? file-root?)
+      (object/raise workspace/current-ws :rename! path neue)
+      (object/raise workspace/current-ws :watched.rename path neue))
+    (files/move! path neue)
+    (object/merge! origin {:path neue})
+    (remove-child (find-by-path old-parent-path) origin)
+    (add-child (find-by-path new-parent-path) origin)))
+
+(defn rename-folder
```

Comment:
Right around here would be the relevant rename-folder code, in case you wanted to check it regarding the above issue, @kenny-evitt